numba/cpython/listobj.py

```diff
@@ -921,57 +940,35 @@ def list_extend(lst, iterable):
 
     return context.compile_internal(builder, list_extend, sig, args)
 
-@lower_builtin("list.index", types.List, types.Any)
-def list_index(context, builder, sig, args):
 
-    def list_index_impl(lst, value):
-        for i in range(len(lst)):
-            if lst[i] == value:
-                return i
-        # XXX references are leaked when raising
-        raise ValueError("value not in list")
+intp_max = np.iinfo(as_dtype(types.intp)).max
```

Comment:
If possible, the `cpython` implementations should not rely on or refer to NumPy. In this case I think the `intp_max` can be found via:
```
>>> from numba import types
>>> types.intp.maxval
9223372036854775807
```